src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -42,6 +43,205 @@ def get_server_location(server_name, resource_group_name):
 
 
 ###############################################
+#                sql                          #
+###############################################
+
+
+def capabilities_get(  # pylint: disable=too-many-arguments
+        client,
+        location_id,
+        status=None,
+        edition=None,
+        service_objective=None,
+        depth=None):
+
+    # For reference, below is the tree structure of location capabilities:
+    # - location
+    #     - supportedServerVersions
+    #         - supportedEditions
+    #             - supportedServiceLevelObjectives
+    #                 - supportedMaxSizes
+    #         - supportedElasticPoolEditions
+    #             - supportedElasticPoolDtus
+    #                 - supportedMaxSizes
+    #                 - supportedPerDatabaseMaxDtus
+    #                     - supportedPerDatabaseMinDtus
+    #                 - supportedPerDatabaseMaxSizes
+
+    # Convert inputs to correct type
+    status = CapabilityStatus(status) if status else CapabilityStatus.visible
+    depth = int(depth) if depth is not None else 999
```

Comment:
It might make sense to use the shallowest practical depth by default and force the user to opt-in to more verbose output rather than opt-out.